src/utils/markets.ts

```diff
@@ -0,0 +1,41 @@
+import { SavedMarket } from 'src/types';
+
+const SAVED_MARKETS_KEY = 'SAVED_MARKETS_KEY';
+
+export function writeMarketInStore(_market: SavedMarket) {
+    const savedMarkets = JSON.parse(
+        localStorage.getItem(SAVED_MARKETS_KEY) || '[]'
+    );
+    savedMarkets.push(_market);
+    localStorage.setItem(SAVED_MARKETS_KEY, JSON.stringify(savedMarkets));
+}
+
+export function readMarketsFromStore(): SavedMarket[] {
+    const savedMarkets = localStorage.getItem(SAVED_MARKETS_KEY);
+    return savedMarkets ? JSON.parse(savedMarkets) : [];
+}
+
+export function removeMarketFromStore(_market: SavedMarket) {
+    const savedMarkets = JSON.parse(
+        localStorage.getItem(SAVED_MARKETS_KEY) || '[]'
+    );
+    const updatedMarkets = savedMarkets.filter(
+        (savedMarket: SavedMarket) =>
+            savedMarket.market !== _market.market ||
```

Comment:
shouldn't these conditions be and?
even if one of them is true, the _market will not match.

Reply:
No it shouldn't be. A market is kept in the list if it differs from the input market in at least one of these aspects: the market name, the address, or the chain ID. If a saved market matches the input market in all these aspects, it's excluded from the new list.